odl/contrib/fom/supervised.py

```diff
@@ -15,12 +15,13 @@
 
 __all__ = ('mean_squared_error', 'mean_absolute_error',
            'mean_value_difference', 'standard_deviation_difference',
-           'range_difference', 'blurring', 'false_structures', 'ssim', 'psnr',
-           'haarpsi', 'noise_power_spectrum')
+           'range_difference', 'blurring', 'false_structures_mask', 'ssim',
+           'psnr', 'haarpsi', 'noise_power_spectrum')
 
 
-def mean_squared_error(data, ground_truth, mask=None, normalized=False):
-    """Return mean squared L2 distance between ``data`` and ``ground_truth``.
+def mean_squared_error(data, ground_truth, mask=None,
+                       normalized=False, force_lower_is_better=False):
```

Comment:
I think that the default value for this flag should always be false because that’s what one usually would expect when calling a function like SSIM where `force_lower_is_better=True` would reverse the sign.

Reply:
Sure, but why shouldn't different FOMs have different defaults?

Reply:
Actually, the whole reason for this oddity is the prefix `force_`. It makes the parameter a flag for *doing something* vs *doing nothing*. This is a much narrower construct than a flag for *doing one thing* vs *doing another thing*.
If the parameter was called `lower_is_better`, it would be quite natural to set its default value to `False` for SSIM and other measures that are *scores*, and to set it to `True` for *error measures*. And still you could enforce uniform behavior by choosing one of them across all FOMs.

Reply:
This flag is meant to *change* the default behaviour, but perhaps this is only a matter of proper documentation (which is really an issue: http://odlgroup.github.io/odl/odl.contrib.html). Giving a different default behaviour could (in my opinion) suggest setting this flag changes anything for FOMS which have lower values for better quality.

Reply:
> This flag is meant to change the default behaviour

Yes, and that's a one-way operation currently because there's only one direction into which to change. With my suggestion, the default value *indicates* the default behavior, and you could change it in either direction. IMO that would be a much more intuitive interface.



> Giving a different default behaviour could (in my opinion) suggest setting this flag changes anything for FOMs which have lower values for better quality.

Yes, and I think it should.